main/nav_backplane_viewer.py

```
@@ -0,0 +1,1365 @@
+                self._fits_hdus = list(hdul)
+                self._logger.info('Opened backplanes FITS file "%s" with %d HDUs',
+                                  local_path, len(self._fits_hdus))
+                # Parse HDUs: locate BODY_ID_MAP and backplanes
+                for hdu in self._fits_hdus[1:]:
+                    name = (hdu.name or '').upper()
+                    if name == 'BODY_ID_MAP':
+                        try:
+                            self._body_id_map = np.asarray(hdu.data, dtype=np.int32)
+                        except Exception:
+                            self._logger.exception('Failed to parse BODY_ID_MAP from HDU '
+                                                   f'for {fits_file}')
```

Ruff (0.14.5)

Logging statement uses f-string

(G004)